algoliasearch/check_synonym.go

```diff
@@ -0,0 +1,27 @@
+package algoliasearch
+
+import "fmt"
+
+func checkSynonym(s Synonym) error {
+	switch s := s.(type) {
+	case AltCorrectionSynonym,
```

Comment:
Right. I fixed it by allowing `altCorrection1|altCorrection2` to be a possible `Type` string value for the `AltCorrectionSynonym` structure.